services/management/dbaas/psmdb_cluster_service_test.go

```diff
@@ -0,0 +1,233 @@
+				},
+				"name": "self-hosted-cluster"
+			}
+		],
+		"contexts": [
+			{
+				"context": {
+					"cluster": "self-hosted-cluster",
+					"user": "percona-xtradb-cluster-operator"
+				},
+				"name": "svcs-acct-context"
+			}
+		],
+		"current-context": "svcs-acct-context"
+	}
+`
+const kubernetesClusterNameTest = "test-k8s-cluster-name"
+
+func TestPSMDBClusterService(t *testing.T) {
+	setup := func(t *testing.T) (ctx context.Context, db *reform.DB, dbaasClient *mockDbaasClient, teardown func(t *testing.T)) {
+		t.Helper()
+
+		ctx = logger.Set(context.Background(), t.Name())
+		uuid.SetRand(new(tests.IDReader))
+
+		sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+		db = reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+		dbaasClient = new(mockDbaasClient)
+
+		teardown = func(t *testing.T) {
+			uuid.SetRand(nil)
+			dbaasClient.AssertExpectations(t)
+		}
+
+		return
+	}
+
+	ctx, db, dbaasClient, teardown := setup(t)
+	defer teardown(t)
+
+	ks := NewKubernetesServer(db, dbaasClient)
+	dbaasClient.On("CheckKubernetesClusterConnection", ctx, kubeconfTest).Return(nil)
+
+	registerKubernetesClusterResponse, err := ks.RegisterKubernetesCluster(ctx, &dbaasv1beta1.RegisterKubernetesClusterRequest{
+		KubernetesClusterName: kubernetesClusterNameTest,
+		KubeAuth:              &dbaasv1beta1.KubeAuth{Kubeconfig: kubeconfTest},
+	})
+	require.NoError(t, err)
+	assert.NotNil(t, registerKubernetesClusterResponse)
+
+	t.Run("BasicListPSMDBClusters", func(t *testing.T) {
+		s := NewPSMDBClusterService(db, dbaasClient)
+		mockResp := controllerv1beta1.ListPSMDBClustersResponse{
+			Clusters: []*controllerv1beta1.ListPSMDBClustersResponse_Cluster{
+				{
+					Name: "first.pxc.test.percona.com",
+					Params: &controllerv1beta1.PSMDBClusterParams{
+						ClusterSize: 5,
+						Replicaset: &controllerv1beta1.PSMDBClusterParams_ReplicaSet{
+							ComputeResources: &controllerv1beta1.ComputeResources{
+								CpuM:        3,
+								MemoryBytes: 256,
+							},
+						},
+					},
+				},
+			},
+		}
+
+		dbaasClient.On("ListPSMDBClusters", ctx, mock.Anything).Return(&mockResp, nil)
+
+		resp, err := s.ListPSMDBClusters(ctx, &dbaasv1beta1.ListPSMDBClustersRequest{KubernetesClusterName: kubernetesClusterNameTest})
+		assert.NoError(t, err)
+		require.NotNil(t, resp.Clusters[0])
+		assert.Equal(t, resp.Clusters[0].Name, "first.pxc.test.percona.com")
+		assert.Equal(t, int32(5), resp.Clusters[0].Params.ClusterSize)
+		assert.Equal(t, int32(3), resp.Clusters[0].Params.Replicaset.ComputeResources.CpuM)
+		assert.Equal(t, int64(256), resp.Clusters[0].Params.Replicaset.ComputeResources.MemoryBytes)
+	})
+
+	//nolint:dupl
+	t.Run("BasicCreatePSMDBClusters", func(t *testing.T) {
+		s := NewPSMDBClusterService(db, dbaasClient)
+		mockReq := controllerv1beta1.CreatePSMDBClusterRequest{
+			KubeAuth: &controllerv1beta1.KubeAuth{
+				Kubeconfig: kubeconfTest,
+			},
+			Name: "third.pxc.test.percona.com",
```

Comment:
Validation should not allow names with dots: 

https://github.com/percona/pmm/blob/PMM-2.0/api/managementpb/dbaas/xtradb_cluster.proto#L156-L164

Reply:
same in other places.

Reply:
oh, missed